realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -45,7 +45,7 @@
  * @see io.realm.Realm
  * @see io.realm.DynamicRealm
  */
-abstract class BaseRealm implements Closeable {
+public abstract class BaseRealm implements Closeable {
```

Comment:
Maybe we do the same as `https://github.com/realm/realm-java/issues/2401#issuecomment-194192660` ? Let `RealmBaseAdapter` take a param of <E extends Realm>? It is kind of reminding user that the life cycle of Realm should be taken care out of the adapter.